platform/src/main/java/org/stellar/anchor/platform/component/share/ObservingAccountsBeans.java

```diff
@@ -2,14 +2,28 @@
 
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
 import org.stellar.anchor.platform.observer.stellar.PaymentObservingAccountStore;
 import org.stellar.anchor.platform.observer.stellar.PaymentObservingAccountsManager;
 
 @Configuration
 public class ObservingAccountsBeans {
+  private final Environment env;
+
+  public ObservingAccountsBeans(Environment env) {
+    this.env = env;
+  }
+
   @Bean
```

Comment:
Does `@ConditionalOnAnySepsEnabled(seps = {"sep31"})` work? We use this annotation to conditionally create other dependencies.

Reply:
The issue is we need this bean even when sep31 is disabled because `PaymentObserverBeans ` needs it. So that leads to the choice that we always create this bean but not activate it

Reply:
When SEP-31 is disabled, the account the payment observer is monitoring is the distribution account. That is why the eviction task does not need to be enabled when SEP-31 is disabled because it will never be evicted. Is this correct?

Reply:
That's right